Frameworks/Accelerate/vImage.cpp

```diff
@@ -951,4 +935,876 @@ vImage_Error vImageMatrixMultiply_ARGB8888(const vImage_Buffer* src,
+
+                // Pack and saturate to uint8
+                vInt8 = _mm_packus_epi16(vInt16[0], vInt16[1]);
+
+                // Mask out invalid alpha from output pixels
+                vInt8 = _mm_and_si128(vInt8, vInt8ColorMask);
+
+                // Combine with original alpha
+                vInt8 = _mm_or_si128(vInt8, vInt8Alphas);
+
+                // Write pixels to memory
+                _mm_storeu_si128(reinterpret_cast<__m128i*>(&destPixels[offset]), vInt8);
+
+                // Increment pixel offset
+                offset += pixelsPerBlock;
+            }
+
+            srcRowStart += srcRowPitch;
+            destRowStart += destRowPitch;
+        }
+    } else {
+#endif
+        for (unsigned int i = 0; i < height; i++) {
+            srcPixels = reinterpret_cast<Pixel_8888_s*>(srcRowStart);
+            destPixels = reinterpret_cast<Pixel_8888_s*>(destRowStart);
+
+            for (unsigned int j = 0; j < width; j++) {
+                _vImageUnpremultiplyPixel<alphaPos>(&destPixels[j], &srcPixels[j]);
+            }
+
+            srcRowStart += srcRowPitch;
+            destRowStart += destRowPitch;
+        }
+    }
+
+    return kvImageNoError;
+}
+
+/// Unpremultiply pixel
+template <int alphaPos>
+inline void _vImageUnpremultiplyPixel(Pixel_8888_s* dest, Pixel_8888_s* src);
+
+template <>
+inline void _vImageUnpremultiplyPixel<0>(Pixel_8888_s* dest, Pixel_8888_s* src) {
+    dest->val[0] = src->val[0];
+    dest->val[1] = _vImageDivideAndSaturateUint8(src->val[1], src->val[0]);
+    dest->val[2] = _vImageDivideAndSaturateUint8(src->val[2], src->val[0]);
+    dest->val[3] = _vImageDivideAndSaturateUint8(src->val[3], src->val[0]);
+}
+
+template <>
+inline void _vImageUnpremultiplyPixel<3>(Pixel_8888_s* dest, Pixel_8888_s* src) {
+    dest->val[3] = src->val[3];
+    dest->val[0] = _vImageDivideAndSaturateUint8(src->val[0], src->val[3]);
+    dest->val[1] = _vImageDivideAndSaturateUint8(src->val[1], src->val[3]);
+    dest->val[2] = _vImageDivideAndSaturateUint8(src->val[2], src->val[3]);
+}
+
+/**
+@Status Interoperable
+@Notes Packed buffer byte order notation is little endian
+*/
+vImage_Error vImageUnpremultiplyData_RGBA8888(const vImage_Buffer* src, const vImage_Buffer* dest, vImage_Flags flags) {
+    return _vImageUnpremultiplyData_8888<3>(src, dest, flags);
+}
+
+/**
+@Status Interoperable
+@Notes Packed buffer byte order notation is little endian
+*/
+vImage_Error vImageUnpremultiplyData_BGRA8888(const vImage_Buffer* src, const vImage_Buffer* dest, vImage_Flags flags) {
+    return _vImageUnpremultiplyData_8888<3>(src, dest, flags);
+}
+
+/**
+@Status Interoperable
+@Notes Packed buffer byte order notation is little endian
+*/
+vImage_Error vImageUnpremultiplyData_ARGB8888(const vImage_Buffer* src, const vImage_Buffer* dest, vImage_Flags flags) {
+    return _vImageUnpremultiplyData_8888<0>(src, dest, flags);
+}
+
+/**
+@Status Interoperable
+*/
+vImage_Error vImageBuffer_Init(
```

Comment:
This is an API that does not appear in the official documentation but is certainly part of the API since many samples use it and it shows up on the iOS headers. Some documentation is available here:
https://developer.apple.com/library/mac/releasenotes/Performance/RN-vecLib/